### ISSUE TITLE
Implementation of user-specified wakefields

### ISSUE THREAD

**Issue by cemitch99** [BLAST-ImpactX/impactx#389]
Allow the user to specify the three components of a single-particle wake function W_z(z), W_x(z), and W_y(z).

May be used to apply longitudinal (monopole) and transverse (dipole) wakefields in an RF or plasma-based accelerating structure.

Requires:
longitudinal binning of the beam distribution to produce lambda(z), x(z)lambda(z), and y(z)lambda(z).
evaluation of 1D FFT-based convolution of the above functions with with W_z, W_x, and W_y, respectively.

**Comment by jonasbjorklundsvensson** [comment]
Hi,

What's the status of this functionality? Whenever I simulate RF linacs, RF wakes are absolutely necessary to include to get even remotely close to a realistic LPS at the end. 

Best,
Jonas

**Comment by cemitch99** [comment]
@jonasbjorklundsvensson  Yes, understood!  Thanks for your comment.  Please be aware that there is a long list of requested features that are required for more immediate internal project milestones, so some Issues have been moved to lower priority.  (On our end, it would be good to more publicly rank Issues by priority, so that users can see where their desired features fall in the queue.)  Also, ImpactX is fully open-source, so any user is welcome to open a pull request for review.

**Comment by jonasbjorklundsvensson** [comment]
@cemitch99 Thanks for the quick reply! I can tell that you are actively developing the project, and I guessed something along the lines of what you're saying. Just thought I'd make my interest known; I'm very interested in using Impactx, but I can't really make the swap until a few key (for me) features are available. I am currently not good enough at coding to make my own contributions, perhaps that will change at some point.

And I think it would indeed be a good idea to have a public ranking of priorities, so that people like me can be directed to it :)
